contents/testing/writing-testable-code.md

```diff
@@ -0,0 +1,110 @@
+# Writing Testable Code
+
+Below is a collection of 10 rules that you can follow to write testable code.
+
+## Don't mix object construction with application logic
+
+Because when you are writing tests, you mostly want to isolate some part of the application.
+
+In order to instantiate the class in isolation you have to make sure that the class itself does not instantiate other objects.
```

Comment:
Ok! Basically for unit testing, ideally there is only one class under testing, that's why we mock out the dependencies. But if you instantiate other objects directly using `new`, that's a dependency we cannot mock out (how to mock out `new`?)